Source/Core/Updater/Main.cpp

```diff
@@ -0,0 +1,681 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <windows.h>
+
+#include <OptionParser.h>
+#include <algorithm>
+#include <array>
+#include <cstdio>
+#include <ed25519/ed25519.h>
+#include <mbedtls/base64.h>
+#include <mbedtls/sha256.h>
+#include <optional>
+#include <shellapi.h>
+#include <vector>
+#include <zlib.h>
+
+#include "Common/CommonPaths.h"
+#include "Common/CommonTypes.h"
+#include "Common/FileUtil.h"
+#include "Common/HttpRequest.h"
+#include "Common/StringUtil.h"
+
+namespace
+{
+// Public key used to verify update manifests.
+const u8 UPDATE_PUB_KEY[] = {0x2a, 0xb3, 0xd1, 0xdc, 0x6e, 0xf5, 0x07, 0xf6, 0xa0, 0x6c, 0x7c,
+                             0x54, 0xdf, 0x54, 0xf4, 0x42, 0x80, 0xa6, 0x28, 0x8b, 0x6d, 0x70,
+                             0x14, 0xb5, 0x4c, 0x34, 0x95, 0x20, 0x4d, 0xd4, 0xd3, 0x5d};
+
+const char* UPDATE_TEMP_DIR = "TempUpdate";
+
+// Where to log updater output.
+FILE* log_fp = stderr;
+
+void FlushLog()
+{
+  fflush(log_fp);
+  fclose(log_fp);
+}
+
+// Internal representation of options passed on the command-line.
+struct Options
+{
+  std::string this_manifest_url;
+  std::string next_manifest_url;
+  std::string content_store_url;
+  std::string install_base_path;
+  std::optional<DWORD> parent_pid;
+  std::optional<std::string> log_file;
+};
+
+std::vector<std::string> CommandLineToUtf8Argv(PCWSTR command_line)
+{
+  int nargs;
+  LPWSTR* tokenized = CommandLineToArgvW(command_line, &nargs);
```

Comment:
```cpp
std::unique_ptr<LPWSTR, decltype(&LocalFree)> tokenized{CommandLineToArgvW(command_line, &nargs), LocalFree};
```

can be used here to avoid janitoring the LocalFree at the end of the function, though it does look a little noisier, so I can get if this was avoided intentionally.

Reply:
Can't because the types don't exactly match (HLOCAL, etc.).